runtime/drivers/snowflake/sql_store.go

```
@@ -0,0 +1,250 @@
+	"github.com/rilldata/rill/runtime/pkg/observability"
+	sf "github.com/snowflakedb/gosnowflake"
+	"go.uber.org/zap"
+)
+
+// recommended size is 512MB - 1GB, entire data is buffered in memory before its written to disk
+const rowGroupBufferSize = int64(datasize.MB) * 512
+
+// Query implements drivers.SQLStore
+func (c *connection) Query(ctx context.Context, props map[string]any) (drivers.RowIterator, error) {
+	return nil, drivers.ErrNotImplemented
+}
+
+// QueryAsFiles implements drivers.SQLStore.
+// Fetches query result in arrow batches.
+// As an alternative (or in case of memory issues) consider utilizing Snowflake "COPY INTO <location>" feature,
+// see https://docs.snowflake.com/en/sql-reference/sql/copy-into-location
+func (c *connection) QueryAsFiles(ctx context.Context, props map[string]any, opt *drivers.QueryOption, p drivers.Progress) (drivers.FileIterator, error) {
+	srcProps, err := parseSourceProperties(props)
+	if err != nil {
+		return nil, err
+	}
+
+	var dsn string
+	if srcProps.DSN != "" { // get from src properties
+		dsn = srcProps.DSN
+	} else if url, ok := c.config["dsn"].(string); ok && url != "" { // get from driver configs
+		dsn = url
+	} else {
+		return nil, fmt.Errorf("the property 'dsn' is required for Snowflake. Provide 'dsn' in the YAML properties or pass '--env connector.snowflake.dsn=...' to 'rill start'")
+	}
+
+	db, err := sql.Open("snowflake", dsn)
+	if err != nil {
+		log.Fatalf("failed to connect. %v, err: %v", dsn, err)
+	}
+
+	ctx = sf.WithOriginalTimestamp(sf.WithArrowAllocator(sf.WithArrowBatches(ctx), memory.DefaultAllocator))
+
+	conn, err := db.Conn(ctx)
+	if err != nil {
+		db.Close()
+		return nil, err
+	}
+
+	var rows sqld.Rows
+	err = conn.Raw(func(x interface{}) error {
+		rows, err = x.(sqld.QueryerContext).QueryContext(ctx, srcProps.SQL, nil)
+		return err
+	})
+	if err != nil {
+		rows.Close()
+		conn.Close()
+		db.Close()
+		log.Fatalf("unable to run the query. err: %v", err)
```

This should return error instead of `log.Fatalf` which exits the program.